src/impl_/pyclass.rs

```diff
@@ -1102,6 +1102,13 @@ impl<T> PyClassThreadChecker<T> for ThreadCheckerImpl {
 }
 
 /// Trait denoting that this class is suitable to be used as a base type for PyClass.
+
+#[cfg_attr(
+    all(diagnostic_namespace, feature = "abi3"),
+    diagnostic::on_unimplemented(
+        note = "with the `abi3` feature enabled native types cannot be subclassed"
```

Comment:
It might be worth making it clear this is a technical deficiency in PyO3 which we might overcome one day:

```suggestion
        note = "with the `abi3` feature enabled PyO3 cannot yet subclass native types"
```

Reply:
I've gone with `"PyO3 does not support subclassing native types"` because I feel "cannot yet" implies that it's coming soon. I'm not aware of any work being done on that.

Reply:
Sounds great to me! This is on the list of "things I'd like to tackle" but realistically is below 3.13 and free threading in priority now.